File: JAVA/S2_SORTING/L1_BASIC/bubblesort.java
import java.util.ArrayList;
import java.util.Scanner;
class sort{
    public void bubblesort(ArrayList<Integer>arr){
        for(int i=0;i<arr.size();i++){
            for(int j=0;j<arr.size()-i-1;j++){
                if(arr.get(j)>arr.get(j+1)){
                    int t=arr.get(j);
                    arr.set(j,arr.get(j+1));
                    arr.set(j+1,t);
                }
            }
        }
    }
}
public class bubblesort{
    public static void main(String[]args){
        sort s=new sort();
        Scanner sc=new Scanner(System.in);
        ArrayList<Integer> arr=new ArrayList<>();
        System.out.print("enter size of array: ");
        int n=sc.nextInt();
        System.out.print("enter element of array: ");
        for(int i=0;i<n;i++){
            arr.add(sc.nextInt());
        }
        s.bubblesort(arr);
        for(int i=0;i<n;i++){
            System.out.print(arr.get(i)+" ");
        }

    }
}

File: CPP/S2_SORTING/L1_BASIC/bubblesort.cpp
#include<iostream>
#include<vector>
using namespace std;
void bubblesort(vector<int>&v){
   int n=v.size();
   for(int i=0;i<n-1;i++){
      for(int j=0;j<n-i-1;j++){
         if(v[j]>v[j+1]){
            int t=v[j];
            v[j]=v[j+1];
            v[j+1]=t;
         }
      }
   }
}
int main(){
   vector<int>arr={23,12,34,89,21};
   bubblesort(arr);
   for(int i:arr){
      cout<<i<<" ";
   }
   return 0;
}